shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -1529,12 +1550,16 @@ public void process(ReplyProcessor cp) {
         public void writeFields(DataOutput out) throws IOException {
             out.writeLong(requestId);
             out.writeUTF(serverStackTrace);
+            out.writeInt(errorCode);
         }
 
         public static WireCommand readFrom(ByteBufInputStream in, int length) throws IOException {
             long requestId = in.readLong();
             String serverStackTrace = (in.available() > 0) ? in.readUTF() : EMPTY_STACK_TRACE;
-            return new AuthTokenCheckFailed(requestId, serverStackTrace);
+
+            // errorCode is a new field and wasn't present earlier. Doing this to allow it to work with older clients.
+            int errorCode = in.available()  >= Integer.BYTES ? in.readInt() : -1;
```

Comment:
As this is adding a new field, you should bump the version number of the protocol incase at some point we want to make this mandatory, we can add that number as the lower bound. (We shouldn't do that now, but we should increase the constant at the top of the file)

Reply:
Sure, done.